Implement the in-game load game menu

TomWerner opened the pull request:
I'm pretty sure this isn't the right way to do this, since the load game menu isn't centered or the right size, like it was after #497. However, I think the fact that the `GameMenu` class is an anonymous node (the path is something like `/root/C7Game/CanvasLayer/PopupOverlay/@TextureRect@350"`, where the number changes run-to-run) means that the approach from #497 doesn't work. Maybe more things need to become their own small scenes? I don't know enough about godot to say for sure.

This seems like an improvement though, since a new button works, but I'd welcome any advice on how to do this the "correct" way.

TomWerner committed:
Implement the in-game load game menu

I'm pretty sure this isn't the right way to do this, since the load game menu isn't centered or the right size, like it was after #497. However, I think the fact that the `GameMenu` class is an anonymous node (the path is something like `/root/C7Game/CanvasLayer/PopupOverlay/@TextureRect@350"`, where the number changes run-to-run) means that the approach from #497 doesn't work. Maybe more things need to become their own small scenes? I don't know enough about godot to say for sure.

This seems like an improvement though, since a new button works, but I'd welcome any advice on how to do this the "correct" way.

esbudylin commented:
My advice is to create an instance of the 'civ3_file_dialog.tscn' scene within the C7Game scene. This way, you will be able to reuse the file dialog's UI appearance. You can instantiate it either in the editor (I think this is the preferable option, as it makes the scene's structure more discoverable) or via code.

esbudylin committed:
Merge branch 'Development' into twrner/in-game-load-game-button

esbudylin committed:
Load menu improvements

- Centered the in-game load menu.
- Moved save file loading logic to Civ3FileDialog to reduce duplication.
- Removed an unclear and likely outdated comment from Civ3FileDialog.
- The popup menu now disappears after opening the load menu, similar to Civ3.
